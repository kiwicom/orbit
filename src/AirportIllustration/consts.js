// @flow

// eslint-disable-next-line import/prefer-default-export
export const NAMES = [
  "BGYFastTrack",
  "BUDFastTrack",
  "MRSSmartPass",
  "NCEFastTrack",
  "PRGSmartPass",
  "VCESmartPass",
];
