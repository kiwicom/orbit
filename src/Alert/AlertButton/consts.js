// @flow
export const TYPE_OPTIONS = {
  INFO: "info",
  SUCCESS: "success",
  WARNING: "warning",
  CRITICAL: "critical",
  INFO_SUBTLE: "infoSubtle",
  SUCCESS_SUBTLE: "successSubtle",
  WARNING_SUBTLE: "warningSubtle",
  CRITICAL_SUBTLE: "criticalSubtle",
};

export const TOKENS = {
  backgroundButton: "backgroundButton",
  backgroundButtonHover: "backgroundButtonHover",
  backgroundButtonActive: "backgroundButtonActive",
  backgroundButtonFocus: "backgroundButtonFocus",
  colorTextButton: "colorTextButton",
  borderColorButtonFocus: "borderColorButtonFocus",
};
