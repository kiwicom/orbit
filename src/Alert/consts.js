// @flow
export const TYPE_OPTIONS = {
  INFO: "info",
  SUCCESS: "success",
  WARNING: "warning",
  CRITICAL: "critical",
};

export const TOKENS = {
  backgroundAlert: "backgroundAlert",
  colorIconAlert: "colorIconAlert",
  colorTextAlert: "colorTextAlert",
  colorTextLinkAlertHover: "colorTextLinkAlertHover",
  colorTextLinkAlertFocus: "colorTextLinkAlertFocus",
  colorBorderAlert: "colorBorderAlert",
};

export const CLOSE_BUTTON_DATA_TEST = "AlertCloseButton";
