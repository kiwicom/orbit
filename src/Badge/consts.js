// @flow
export const TYPE_OPTIONS = {
  NEUTRAL: "neutral",
  INFO: "info",
  SUCCESS: "success",
  WARNING: "warning",
  CRITICAL: "critical",
  DARK: "dark",
  WHITE: "white",
  INFO_INVERTED: "infoInverted",
  CRITICAL_INVERTED: "criticalInverted",
};

export const TOKENS = {
  background: "background",
  color: "color",
};
