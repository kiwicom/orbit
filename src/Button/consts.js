// @flow
export const TYPE_OPTIONS = {
  PRIMARY: "primary",
  SECONDARY: "secondary",
  INFO: "info",
  SUCCESS: "success",
  WARNING: "warning",
  CRITICAL: "critical",
  FACEBOOK: "facebook",
  GOOGLE: "google",
  WHITE: "white",
};

export const SIZE_OPTIONS = {
  SMALL: "small",
  NORMAL: "normal",
  LARGE: "large",
};

export const TOKENS = {
  // Size tokens
  heightButton: "heightButton",
  loadingWidth: "loadingWidth",
  loadingHeight: "loadingHeight",
  fontSizeButton: "fontSizeButton",
  paddingButton: "paddingButton",
  paddingButtonWithIcons: "paddingButtonWithIcons",
  paddingButtonWithLeftIcon: "paddingButtonWithLeftIcon",
  paddingButtonWithRightIcon: "paddingButtonWithRightIcon",
  marginRightIcon: "marginRightIcon",
  // Type tokens
  backgroundButton: "backgroundButton",
  backgroundButtonHover: "backgroundButtonHover",
  backgroundButtonActive: "backgroundButtonActive",
  backgroundButtonBordered: "backgroundButtonBordered",
  backgroundButtonBorderedHover: "backgroundButtonBorderedHover",
  backgroundButtonBorderedActive: "backgroundButtonBorderedActive",
  colorTextButton: "colorTextButton",
  colorTextButtonBordered: "colorTextButtonBordered",
  colorTextButtonHover: "colorTextButtonHover",
  colorTextButtonBorderedHover: "colorTextButtonBorderedHover",
  colorTextButtonActive: "colorTextButtonActive",
  colorTextButtonBorderedActive: "colorTextButtonBorderedActive",
  borderColorButton: "borderColorButton",
  borderColorButtonHover: "borderColorButtonHover",
  borderColorButtonActive: "borderColorButtonActive",
};

export const BUTTON_STATES = {
  DEFAULT: "default",
  HOVER: "hover",
  ACTIVE: "active",
  FOCUS: "focus",
};
