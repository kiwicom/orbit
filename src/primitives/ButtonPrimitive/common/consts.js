// @flow
export const SIZE_OPTIONS = {
  SMALL: "small",
  NORMAL: "normal",
  LARGE: "large",
};

export const TOKENS = {
  heightButton: "heightButton",
  fontSizeButton: "fontSizeButton",
  paddingButton: "paddingButton",
  paddingButtonWithIcons: "paddingButtonWithIcons",
  paddingButtonWithLeftIcon: "paddingButtonWithLeftIcon",
  paddingButtonWithRightIcon: "paddingButtonWithRightIcon",
  spinnerWidth: "spinnerWidth",
  spinnerHeight: "spinnerHeight",
  marginRightIcon: "marginRightIcon",
};

export const BUTTON_STATES = {
  DEFAULT: "default",
  HOVER: "hover",
  ACTIVE: "active",
  FOCUS: "focus",
};
