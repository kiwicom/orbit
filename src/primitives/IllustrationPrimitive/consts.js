// @flow

export const SIZE_OPTIONS = {
  SMALL: "small",
  MEDIUM: "medium",
  LARGE: "large",
  DISPLAY: "display",
};

export const baseURL = "//images.kiwi.com";
