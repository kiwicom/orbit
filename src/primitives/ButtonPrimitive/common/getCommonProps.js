// @flow
import getSizeToken from "./getSizeToken";
import getPadding from "./getPadding";
import type { GetCommonProps } from "./getCommonProps";
import { SIZE_OPTIONS } from "./consts";

const getCommonProps: GetCommonProps = ({
  width,
  size = SIZE_OPTIONS.NORMAL,
  theme,
  iconRight,
  iconLeft,
  children,
}) => {
  const onlyIcon = Boolean(iconLeft && !children);
  return {
    ...getSizeToken(size, theme),
    width,
    padding: getPadding(onlyIcon, iconRight, iconLeft, size, theme),
    fontWeight: theme.orbit.fontWeightBold,
  };
};

export default getCommonProps;
