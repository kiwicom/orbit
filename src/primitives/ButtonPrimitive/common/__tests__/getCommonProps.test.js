// @flow
import getCommonProps from "../getCommonProps";
import theme from "../../../../defaultTheme";
import { SIZE_OPTIONS } from "../consts";

describe("ButtonPrimitive: getCommonProps function", () => {
  it("should match: null width and static fontWeight and transition", () => {
    const size = SIZE_OPTIONS.LARGE;
    const onlyIcon = false;
    const iconLeft = true;
    expect(getCommonProps({ size, onlyIcon, iconLeft, theme })).toMatchSnapshot();
  });
  it("should match: dynamic width and static fontWeight and transition", () => {
    const width = "200px";
    const size = SIZE_OPTIONS.NORMAL;
    const onlyIcon = true;
    const iconLeft = true;
    expect(
      getCommonProps({
        width,
        size,
        onlyIcon,
        iconLeft,
        theme,
      }),
    ).toMatchSnapshot();
  });
});
