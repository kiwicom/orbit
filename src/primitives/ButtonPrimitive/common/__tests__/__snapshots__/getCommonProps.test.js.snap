// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPrimitive: getCommonProps function should match: dynamic width and static fontWeight and transition 1`] = `
Object {
  "fontSize": "14px",
  "fontWeight": "700",
  "height": "44px",
  "padding": "0",
  "width": "200px",
}
`;

exports[`ButtonPrimitive: getCommonProps function should match: null width and static fontWeight and transition 1`] = `
Object {
  "fontSize": "16px",
  "fontWeight": "700",
  "height": "52px",
  "padding": "0",
  "width": undefined,
}
`;
